src/components/TodoItem.tsx

```diff
@@ -0,0 +1,134 @@
+import { Dispatch, SetStateAction, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+
+type FormEventType =
+  | React.FormEvent<HTMLFormElement>
+  | React.FocusEvent<HTMLFormElement, Element>;
+
+type Props = {
+  todo: Todo;
+  handleDeleteTodo: (todoId: number) => Promise<void>;
+  isLoading?: boolean;
+  handleUpdateTodo: (todoToUpdate: Todo) => Promise<void>;
+  isInEditMode?: boolean;
+  setEditedTodoId: Dispatch<SetStateAction<number | null>>;
+};
+
+export const TodoItem: React.FC<Props> = props => {
+  const {
+    todo,
+    isLoading,
+    handleDeleteTodo,
+    handleUpdateTodo,
+    isInEditMode,
+    setEditedTodoId,
+  } = props;
+
+  const [todoTitleValue, setTodoTitleValue] = useState<string>(todo.title);
```

Comment:
if u use (string, number, boolean) as the initial value - u don't need a generic type
```suggestion
  const [todoTitleValue, setTodoTitleValue] = useState(todo.title);
```